### PR TITLE
[Keyboard] Add SP-111 v2 keyboard

### PR DESCRIPTION
**Pull request by thompson-ele**
## Description

Added SP-111 keyboard v2 PCB support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/viktus/sp111_v2/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,                            _______, _______, _______)
+};
+
+#ifdef RGB_MATRIX_ENABLE
```
Comment (zvecr) [inline keyboards/viktus/sp111_v2/keymaps/via/keymap.c:32]: Unfortunately, custom code for indicators is no longer being accepted to the repo and will have to be removed for the PR to be accepted.

File: keyboards/viktus/sp111_v2/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, QK_BOOT,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,                            _______, _______, _______)
```
Comment (dunk2k) [inline keyboards/viktus/sp111_v2/keymaps/via/keymap.c:29]: non-functional layer removed
```suggestion
    _______, _______, _______, _______,      _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,                            _______, _______, _______)
```

File: keyboards/viktus/sp111_v2/keymaps/via/config.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2023 BlindAssassin111 (@blindassassin111)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (dunk2k) [inline keyboards/viktus/sp111_v2/keymaps/via/config.h:3]:
```suggestion
#pragma once

```

File: keyboards/viktus/sp111_v2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, QK_BOOT,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+    _______, _______, _______, _______,      _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,                            _______, _______, _______),
```
Comment (dunk2k) [inline keyboards/viktus/sp111_v2/keymaps/default/keymap.c:48]:
```suggestion
    _______, _______, _______, _______,      _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,                            _______, _______, _______)
```

File: keyboards/viktus/sp111_v2/info.json [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    },
+    "rgb_matrix": {
+        "driver": "WS2812",
+        "max_brightness": 150,
+        "layout": [
+        ]
+    },
+    "split": {
+        "enabled": true,
+        "main": "left",
```
Comment (dunk2k) [inline keyboards/viktus/sp111_v2/info.json:36]: 1. `main` currently not working, fix under way. work around is to define handedness in `config.h`
2. MASTER_LEFT is default handedness setting for split keyboards
```suggestion
```

File: keyboards/viktus/sp111_v2/sp111_v2.h
Comment (dunk2k) [inline keyboards/viktus/sp111_v2/sp111_v2.h:1]: could you remove this file as matrix positions have been defined in `layouts.LAYOUT.layout` of `info.json`

File: keyboards/viktus/sp111_v2/config.h
Comment (dunk2k) [inline keyboards/viktus/sp111_v2/config.h:1]: Please migrate the contents of this file to `info.json` and then remove file

File: keyboards/viktus/sp111_v2/info.json [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    "keyboard_folder": "viktus/sp111_v2",
+    "keymaps": {},
+    "matrix_pins": {
+        "cols": ["C6", "F6", "F5", "F4", "F1", "F7", "B7", "D2", "D3", "D5", "D4"],
+        "rows": ["C7", "B6", "B5", "B4", "D7", "D6"]
+    },
+    "matrix_size": {
+        "cols": 11,
+        "rows": 12
+    },
```
Comment (dunk2k) [inline keyboards/viktus/sp111_v2/info.json:27]: not required
```suggestion
        "rgb_matrix": true
    },
    "matrix_pins": {
        "cols": ["C6", "F6", "F5", "F4", "F1", "F7", "B7", "D2", "D3", "D5", "D4"],
        "rows": ["C7", "B6", "B5", "B4", "D7", "D6"]
    },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/viktus/sp111_v2/info.json [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    "rgb_matrix": {
+        "driver": "WS2812",
+        "max_brightness": 150,
+        "layout": [
+        ]
```
[the remark this answers is not in the document]
Comment (thompson-ele) [reply keyboards/viktus/sp111_v2/info.json]: @dunk2k I'm very new to the data driven style of doing the config but I think I moved over what I could. There were a few rgb_matrix related settings I left in config.h that I wasn't entirely sure of how to declare but lmk if you think those could be moved as well :)

File: keyboards/viktus/sp111_v2/keymaps/via/keymap.c (hunk 1 above)
Comment (thompson-ele) [reply keyboards/viktus/sp111_v2/keymaps/via/keymap.c]: I've removed. Thanks

**Review by drashna — APPROVED** [review]

File: keyboards/viktus/sp111_v2/config.h [hunk 8]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 BlindAssassin111 (@blindassassin111)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* Split Defines */
+#define SPLIT_USB_DETECT
+#define MASTER_LEFT
```
Comment (waffle87) [inline keyboards/viktus/sp111_v2/config.h:8]:
```suggestion
```

File: keyboards/viktus/sp111_v2/config.h [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (waffle87) [inline keyboards/viktus/sp111_v2/config.h:33]:
```suggestion
```

File: keyboards/viktus/sp111_v2/config.h [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+#define SPLIT_USB_DETECT
+#define MASTER_LEFT
+
+#ifdef RGB_MATRIX_ENABLE
+    // The number of LEDs connected
+    #define RGB_MATRIX_LED_COUNT 3
+    #define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+    #define RGB_MATRIX_KEYPRESSES
+    #define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+    #define EECONFIG_KB_DATA_SIZE 12
```
Comment (waffle87) [inline keyboards/viktus/sp111_v2/config.h:16]: Is this needed?

File: keyboards/viktus/sp111_v2/config.h [hunk 11]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 BlindAssassin111 (@blindassassin111)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* Split Defines */
+#define SPLIT_USB_DETECT
```
Comment (waffle87) [inline keyboards/viktus/sp111_v2/config.h:7]: Which is it?
https://github.com/thompson-ele/qmk_firmware/blob/38cae8d3adf9ee1c6566a05c666b26d9afbda71f/keyboards/viktus/sp111_v2/info.json#L65-L67
Either way, it should be removed from `config.h` file.
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/viktus/sp111_v2/config.h (hunk 10 above)
Comment (thompson-ele) [reply keyboards/viktus/sp111_v2/config.h]: I've removed the config.h file

File: keyboards/viktus/sp111_v2/info.json [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["C6", "F6", "F5", "F4", "F1", "F7", "B7", "D2", "D3", "D5", "D4"],
+        "rows": ["C7", "B6", "B5", "B4", "D7", "D6"]
+    },
+    "ws2812": {
+        "pin": "F0"
+    },
+    "rgb_matrix": {
+        "driver": "WS2812",
```
Comment (waffle87) [inline keyboards/viktus/sp111_v2/info.json:25]:
```suggestion
        "driver": "ws2812",
```

File: keyboards/viktus/sp111_v2/info.json [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+    "url": "https://viktus.design/",
+    "usb": {
+        "device_version": "2.0.0",
+        "pid": "0x5112",
+        "vid": "0x5644"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+                { "label": "0,0", "matrix": [0, 0], "w": 1, "x": 0, "y": 0 },
```
Comment (fauxpark) [inline keyboards/viktus/sp111_v2/info.json:78]: Default key width and height is 1u.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**